Libraries/Lists/VirtualizedList.js

```diff
@@ -310,8 +313,8 @@ let _usedIndexForKey = false;
 let _keylessItemComponentName: string = '';
 
 type State = {
-  first: number,
-  last: number,
+  renderMask: CellRenderMask,
+  viewportWindow: {first: number, last: number},
```

Comment:
I think we can keep the `viewportWindow` out of state, since it doesn't directly affect render. Only the batch computations used to compute new render masks.

Reply:
No, we still need it for derived state